python/lsst/daf/butler/core/config.py

```diff
@@ -99,6 +105,8 @@ def include(self, node):
             raise yaml.constructor.ConstructorError
 
     def extractFile(self, filename):
+        if self._root is None:
+            raise RuntimeError("Unable to load !include directive from YAML string")
```

Comment:
There's no ticket for fixing this since nothing has tickled this error yet. The problem is mostly associated with configs in S3 which need to then use the URI root to find the include file (which will be relative path). If the `!include` is absolute or you assume local root is `"."` then we could try to continue.  I haven't thought too deeply about the ramifications of this.

Reply:
I've rearranged the code in a new commit. Now I use a relative URI up front if no other context exists and I fixed a but that meant that S3 absolute URIs in `!include` were not working properly (so you couldn't have a local YAML that referred to a remote YAML). Please take another look and see if this makes you happier.